tracing/tests/macros.rs

```diff
@@ -262,6 +262,22 @@ fn error_span_with_parent() {
     error_span!(parent: &p, "bar",);
 }
 
+#[test]
+fn span_with_non_rust_symbol() {
```

Comment:
May be good to add a similar compile test for an event, just in case. This is _probably_ sufficient, since a regression would probably be in the shared key-value parsing macros, so it may not matter a whole lot...

Reply:
We only change valueset and fieldset, not span or event directly, so I think just testing span! is fine, unless things get rearranged radically.